### PR TITLE
refactor: splits templates API into list and fetch by id

### LINKED ISSUE #102: Improve templates API
- The templates API are getting heavy as more templates are added to the awesome akash repo. Also, the endpoint is called many times throughout the UX.
image
---
TODO
- [x] Modifiy the [templates list api endpoint](https://github.com/akash-network/console/blob/main/apps/api/src/routes/v1/templates.ts#L44) to only return the info needed to display in a list (without the readme, etc) with goal to reduce the amount of uncessary data sent to the browser
- [x] Add a new api endpoint for a template detail with all the info

### PR DESCRIPTION
**Pull request by stalniy**
1. Splits templates API into 2: list and getById
2. Reduces amount of properties for every template in response. It contains only 4 properties that is used on UI:
    * id
    * summary
    * logoUrl
    * name

~Not tested.~ Fixes #102

### CODE REVIEW

**Comment by stalniy** [comment]
~DO NOT MERGE it. Because very likely it will break template detail page on UI as UI currently does not fetch template by id from API.~

**Comment by baktun14** [comment]
Thanks for your work!

Sorry if it wasn't in the requirements, but following your comments about the impact on the frontend; I suggest we keep the old `/templates` api endpoint as-is and create a new one `/template-list` with your changes. That way it won't affect the frontend and we'll be able to merge your pr without effect and implement it on the frontend in another pr.

**Comment by stalniy** [comment]
Probably I could create this extra api endpoints under /v2. To make this  evolution more clear from api perspective and at the same time keeping /v1/templates to keep old clients work without issues. Some people also may have cached version of SPA in their browsers (long lived browser tabs), and their version of client app may try to use older API version

**Comment by baktun14** [comment]
> Probably I could create this extra api endpoints under /v2. To make this evolution more clear from api perspective and at the same time keeping /v1/templates to keep old clients work without issues. Some people also may have cached version of SPA in their browsers (long lived browser tabs), and their version of client app may try to use older API version

Great suggestion and I discussed this internally, but for the sake of keeping the scope of this pr small, I suggest we just keep the old version and add the 2 new endpoints you added.

**Comment by stalniy** [comment]
Changed, now it's safe to merge it

File: apps/api/src/routes/v1/templates/list.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+          schema: responseSchema
+        }
+      }
+    }
+  }
+});
+
+export default new OpenAPIHono().openapi(route, async c => {
+  const templatesPerCategory = await getCachedTemplatesGallery();
+  // TODO: remove manual response filtering when https://github.com/honojs/middleware/issues/181 is done
```
Comment (ygrishajev) [inline apps/api/src/routes/v1/templates/list.ts:36]: 🔥

**Review by baktun14 — APPROVED** [review]
Tested it locally, works great. Good job!

**Pull request merged**